File: Problems & Solutions/B/Black and White/09 December 2022/JavaScript/Black and White.js
//{ Driver Code Starts
//Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let input_ar1 = readLine().split(' ').map(x=>parseInt(x));
        let n = input_ar1[0];
        let m = input_ar1[1];
        let obj = new Solution();
        console.log(obj.numOfWays(n, m));
        
    }
}
// } Driver Code Ends

class Solution 
{
    numOfWays(n, m)
    {
        let x_off = [-2, -2, -1, 1, 2, 2, 1, -1];
        let y_off = [-1, 1, 2, 2, 1, -1, -2, -2];
        let MOD = 1000000007;
    
        let ret = 0;
        let x, y;
    
        for (let i = 0; i < n; ++i) 
        {
            for (let j = 0; j < m; ++j) 
            {
                for (let k = 0; k < 8; ++k)
                {
                    x = i + x_off[k];
                    y = j + y_off[k];
                    
                    if (x >= 0 && x < n && y >= 0 && y < m)
                        ++ret; 
                }
            }
        }
        let total = n * m;
        
        total = (total * (total - 1)) % MOD; 
        return (total + MOD - ret) % MOD; 
    }
}

File: Problems & Solutions/B/Black and White/09 December 2022/Java/Black and White.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;
import java.math.BigInteger;

class Driverclass {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int t = 0; t < T; t++) {
            int N = sc.nextInt();
            int M = sc.nextInt();
            System.out.println(new BlackAndWhite().numOfWays(N, M));
        }
    }
}
// } Driver Code Ends

class BlackAndWhite
{
    public static long numOfWays(int n, int m) 
    {
        int x_off[] = new int[]{-2, -2, -1, 1, 2, 2, 1, -1};
        int y_off[] = new int[]{-1, 1, 2, 2, 1, -1, -2, -2};
        final long MOD = 1000000007;

        long ret = 0;
        int x = 0, y = 0;

        for (int i = 0; i < m; ++i)
        {
            for (int j = 0; j < n; ++j) 
            {
                for (int k = 0; k < 8; ++k)
                {
                    x = i + x_off[k];
                    y = j + y_off[k];

                    if (x >= 0 && x < m && y >= 0 && y < n)
                        ++ret; 
                }
            }
        }
        long total = m * n;
        total = (total * (total - 1)) % MOD;
        
        return (total + MOD - ret) % MOD; 
    }

}